docs/developer-guides/transactions/txscript/overview.md

```diff
@@ -0,0 +1,113 @@
+# <img class="dcr-icon" src="/img/dcr-icons/Code.svg" /> txscript
+
+---
+
+Decred uses a scripting system known as
+[txscript](https://github.com/decred/dcrd/tree/master/txscript) for
+transactions.
+txscript is a simple, stack-based,
+[Forth](https://en.wikipedia.org/wiki/Forth_(programming_language))-like
+programming language.
+The language is not Turing complete and does not implement any
+type of looping, but it is still fairly powerful despite this.
+Each of these properties are described in more detail below.
+
+A script can be considered as a list of conditions attached to a transaction,
+which must be met in order for the recipient of a transaction to spend the
+Decred they have received.
+
+In a simple transfer of Decred from one address to another, the script will
+typically specify just two requirements; in order to spend the received the
+Decred, the recipient must provide:
+
+- a public key that, when hashed, yields the receiving address specified in the
+  script
+- a signature to prove ownership of the private key corresponding to that public
+  key
+
+More complex scripts can be constructed requiring multiple private keys, or no
+keys at all, and it is possible to make use of time locks which prevent the
+spending of DCR until some point in the future.
+
```

Comment:
I'd add a paragraph here indicating that in reality multiple scripts are used to determine if the spending of an utxo on some transaction input is valid.

Something like:

```suggestion

The scripts executed by full nodes while validating transactions in a block or before entering the mempool are actually a combination of two or three different scripts: 
  - The _Public Key_ script (`PkScript`) resides in the _previous output_ (the UTXO or in other words, the [output being spent](/developer-guides/transactions/transaction-format/#outputs) by one of the transaction's input).
  - The _Signature Script_ (`SigScript`) is stored in the [_input_](/developer-guides/transactions/transaction-format/#witness-inputs) being validated.
  - The so-called _Redeem Script_ is used on P2SH addresses and is stored as part of the `SigScript` and has some special rules for being detected and executed.
```

>

Reply:
Just to confirm my understanding, is `PkScript` synonymous with `scriptPubKey`?
And `SigScript` synonymous with `scriptSig`?

I notice `scriptPubKey` and `scriptSig` are used in dcrdata when viewing a raw tx ([example](https://explorer.dcrdata.org/api/tx/decoded/ec0115192fc515d2dc67c54fee1c3acadec69070b1db40921037934d4e2ba68c?indent=true)) but I dont think I've seen your terminology anywhere?

wrt.  #87

Reply:
Yes, they're synonymous.  The output field in dcrdata (really it's coming from the underlying RPC output) was inherited and wasn't changed to keep compatibility.

Reply:
Added this section